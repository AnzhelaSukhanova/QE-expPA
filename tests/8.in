(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult (bvmul y #b0011) x) (= (bvmul y #b0011) x)) 
 (or (bvult x (bvmul y #b0110)) (= x (bvmul y #b0110))))))
