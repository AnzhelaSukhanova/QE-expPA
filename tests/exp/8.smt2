(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (or (bvult (bvmul (bvshl #b0001 x) #b1011) (bvadd y #b1100)) (= (bvmul (bvshl #b0001 x) #b1011) (bvadd y #b1100)))))
