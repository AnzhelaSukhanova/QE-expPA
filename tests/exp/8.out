(let (($e1 (bvadd y #b1100))) (let (($e2 (= y #b0100))) (let (($e3 (ite (bvult $e1 #b1000) (ite (bvult $e1 #b0100) (ite (bvult $e1 #b0010) (ite $e2 #b0100 #b0000) #b0001) #b0010) #b0011))) (let (($e4 (bvshl #b1011 (bvadd #b1101 $e3)))) (let (($e5 (bvshl #b1011 (bvadd #b1110 $e3)))) (not (and (not (bvult #b0110 $e1)) (not (bvult #b1011 $e1)) (not (= y #b1010)) (not (= y #b1111)) (not (bvult #b1100 $e1)) (not (= y #b0000)) (not (bvult #b1000 $e1)) (not (= y #b1100)) (not (bvult $e5 $e1)) (not (bvult $e4 $e1)) $e2 (not (= $e1 $e5)) (not (= $e1 $e4)) (not $e2))))))))
